localstack/services/cloudwatch/provider_v2.py

```diff
@@ -479,3 +548,102 @@ def get_metric_statistics(
             )
 
         return GetMetricStatisticsOutput(Datapoints=datapoints, Label=metric_name)
+
+    def _update_state(
+        self,
+        context: RequestContext,
+        alarm: LocalStackAlarm,
+        state_value: str,
+        state_reason: str,
+        state_reason_data: str = None,
+    ):
+        old_state = alarm.alarm["StateValue"]
+        store = self.get_store(context.account_id, context.region)
+        current_time = datetime.datetime.now()
+        store.Histories.append(
+            {
+                "Timestamp": timestamp_millis(alarm.alarm["StateUpdatedTimestamp"]),
+                "HistoryItemType": HistoryItemType.StateUpdate,
+                "AlarmName": alarm.alarm["AlarmName"],
+                "HistoryData": alarm.alarm.get("StateReasonData"),  # FIXME
```

Comment:
Q: what needs to be fixed here? maybe you could add some more details to the note? 🙏

Reply:
We didn't really care for the history data so far and some of the content/formatting is off. Will clarify